src/Tasks/Microsoft.NET.Build.Tasks/CacheArtifactParser.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Globalization;
+using System.Xml.Linq;
+using NuGet.Versioning;
+using NuGet.Packaging.Core;
+
+#if PRODUCT
```

Comment:
It is not as trivial as that, the contents of the file are going to differ across platforms and we will end up with multiple such files and make the test fragile. Logic like https://github.com/dotnet/sdk/blob/b3ee71d0186053f85d0a1d2094e9695ae2e7b1e9/test/Microsoft.NET.Publish.Tests/GivenThatWeWantToCacheAProjectWithDependencies.cs#L156  would need to be written

If it becomes unbearable we should just fork the logic at that point